datadog/data_source_datadog_ip_ranges.go

```diff
@@ -0,0 +1,168 @@
+package datadog
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+	datadog "github.com/zorkian/go-datadog-api"
+)
+
+func dataSourceDatadogIpRanges() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceDatadogIpRangesRead,
+
+		// IP ranges are divided between ipv4 and ipv6
+		Schema: map[string]*schema.Schema{
+			// v4
+			"agents_ipv4": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"api_ipv4": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"apm_ipv4": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"logs_ipv4": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"process_ipv4": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"synthetics_ipv4": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"webhooks_ipv4": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			// v6
+			"agents_ipv6": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"api_ipv6": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"apm_ipv6": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"logs_ipv6": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"process_ipv6": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"synthetics_ipv6": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"webhooks_ipv6": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+		},
+	}
+}
+
+func dataSourceDatadogIpRangesRead(d *schema.ResourceData, meta interface{}) error {
+
+	client := meta.(*datadog.Client)
+
+	ipAddresses, err := client.GetIPRanges()
+
+	if err != nil {
+		return err
+	}
+
+	// v4
+	if len(ipAddresses.Agents["prefixes_ipv4"])+len(ipAddresses.API["prefixes_ipv4"])+len(ipAddresses.Apm["prefixes_ipv4"])+len(ipAddresses.Logs["prefixes_ipv4"])+len(ipAddresses.Process["prefixes_ipv4"])+len(ipAddresses.Synthetics["prefixes_ipv4"])+len(ipAddresses.Webhooks["prefixes_ipv4"]) > 0 {
```

Comment:
So IIUC both IPv4 and IPv6 ranges could be returned in a single request to our API, so I think it should be sufficient to have a single condition checking len of all IPv4 and IPv6. Is there a reason why you did these two separately?

Reply:
In case we would use one _and not_ the other for a given site. e.g. just IPv4 and not IPv6 we would just set an `id` for IPv4. Am I assessing this incorrectly?

Reply:
Well there's only one Id of a given data source and the Id doesn't really matter to the users. If we only used IPv4 and not IPv6 for certain site, we would still have to save *some* id, as some of the lists wouldn't be empty. The value of the Id itself doesn't matter - what matters is whether or not it's set in this case.